src/chains/celo/serializers.ts

```diff
@@ -18,20 +18,32 @@ import {
 import type {
   CeloTransactionSerializable,
   TransactionSerializableCIP42,
+  TransactionSerializableCIP64,
   TransactionSerializedCIP42,
+  TransactionSerializedCIP64,
 } from './types.js'
 
 export const serializeTransactionCelo: SerializeTransactionFn<
   CeloTransactionSerializable
 > = (tx, signature) => {
-  // Handle CIP-42 transactions
-  if (isCIP42(tx))
-    return serializeTransactionCIP42(
-      tx as TransactionSerializableCIP42,
-      signature,
-    )
-
-  // Handle other transaction types
+  if ('maxFeePerGas' in tx && 'maxPriorityFeePerGas' in tx) {
+    // process as CIP42 if any of these fields are present. realistically gatewayfee is not used but is part of spec
+    if (
+      'gatewayFee' in tx ||
+      'gatewayFeeRecipient' in tx ||
+      ('feeCurrency' in tx && tx.type && tx.type === 'cip42')
```

Comment:
I wouldn't remove `isCIP42`, I'd have a `isCIP64` method to hide implementation details here
```ts

if (isCIP4(tx)) {
  return serializeTransactionCIP64(
    tx as TransactionSerializableCIP64,
    signature,
  )
}
if (isCIP42(tx)) {
  ...
}
```